### DISCUSSION TITLE
How to load a custom profile to the triplestore?

### DISCUSSION THREAD

**Discussion by luoenzhen** [RDFLib/prez#312] [Q&A]
>No you can add a custom profile to the triplestore and prez will pick it up on startup. i.e. a copy of the modified ttl file. You will probably need to call it something other than vocpub though, and get the UI to request this modified profile, as I don't think prez will merge the two profiles nicely due to blank nodes.

Hi @recalcitrantsupplant, I tried this method and renamed the profile as http://www.w3.org/ns/dx/prof/Profile (if I understand that readme-dev file correctly) and imported to my triplestore. However, it still doesn't work as expected. Anything I missed?

File attached. (can't upload ttl file, has to rename it to txt)
[customised_profile.ttl.txt](https://github.com/user-attachments/files/17853426/customised_profile.ttl.txt)

**Comment by recalcitrantsupplant** [comment]
Could you confirm this is for version 3?

**Reply by luoenzhen** [reply]
Yes, it is for prez V3.

**Comment by recalcitrantsupplant** [comment]
You want something like this:

```turtle
<https://w3id.org/profile/myCustomProfile>
    a prof:Profile , prez:VocPrezProfile ;
    dcterms:description "Custome Profile to only render specified properties of a SKOS Concept." ;
    dcterms:identifier "custprof"^^xsd:token ;
    dcterms:title "Custom Profile" ;
    altr-ext:constrainsClass
        skos:Concept ;
    altr-ext:hasDefaultResourceFormat "text/anot+turtle" ;
    altr-ext:hasResourceFormat
        "application/ld+json" ,
        "application/rdf+xml" ,
        "text/anot+turtle" ,
        "text/turtle" ;
    altr-ext:hasNodeShape [
        a sh:NodeShape ;
        sh:targetClass skos:Concept ;
        altr-ext:focusToParent skos:inScheme ;
        sh:path your:pred1 , your:pred2 , your:pred3 ;
    ] ;
.
```
Then when you render a concept you want to request that profile e.g. via the query string `?_profile=custprof`
